### DISCUSSION TITLE
About the output format of the logs

### DISCUSSION THREAD

**Discussion by muyuhui** [open-telemetry/opentelemetry-java-instrumentation#11481] [Q&A]
I am using `opentelemetry-javaagent.jar` to export metrics, traces and logs, but the logs exported by `opentelemetry-javaagent.jar` are in json format. Is there any way to modify the exported logs to the format configured in `logback.xml`?

**Comment by laurit** [comment]
> but the logs exported by opentelemetry-javaagent.jar are in json format

agent exports logs in OTLP format. Please elaborate why you believe the logs are exported in json.

> Is there any way to modify the exported logs to the format configured in logback.xml?

no. If you wish to get the logs exactly as written by logback you may need to use some other tool.

**Comment by akorp** [comment]
I have somewhat similar question. So decided to write in this thread instead of opening the new question. I believe the initial question is slightly wrong, as `opentelemetry-javaagent.jar` are not in json, but that is actually what we would like to have. We are trying to have all our logs in json.

In our setup we have java application logs in json format and auto Instrumentation setup for the application with `otel.javaagent`. The issue is that `otel.javaagent` own logs are not in json format (example below). In there any way to set `opentelemetry-javaagent.jar` own logs in json?

>
{"timestamp":"2024-11-17T06:35:26.548Z","version":"1","message":"Commencing graceful shutdown. Waiting for active requests to complete","logger_name":"org.springframework.boot.web.embedded.tomcat.GracefulShutdown","thread_name":"SpringApplicationShutdownHook","level":"INFO","level_value":20000}
[otel.javaagent 2024-11-17 06:35:26:510 +0000] [OkHttp http://otel-otlp-internal-collector.system-opentelemetry-operator.svc.cluster.local:4317/...] WARN io.opentelemetry.exporter.internal.grpc.GrpcExporter - Failed to export spans. Server responded with gRPC status code 2. Error message: Failed to connect to otel-otlp-internal-collector.system-opentelemetry-operator.svc.cluster.local/10.2.12.55:4317

**Reply by trask** [reply]
you could try `-Dotel.javaagent.logging=application`:

https://opentelemetry.io/docs/zero-code/java/agent/configuration/#java-agent-logging-output
